test/e2e/useraccount_test.go

```diff
@@ -0,0 +1,157 @@
+	defer ctx.Cleanup()
+
+	err = ctx.InitializeClusterResources(&framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(t, err, "failed to initialize cluster resources")
+	t.Log("Initialized cluster resources")
+
+	namespace, err := ctx.GetNamespace()
+	require.NoError(t, err, "failed to get namespace where operator needs to run")
+
+	// get global framework variables
+	f := framework.Global
+	client := f.Client.Client
+
+	// wait for operator to be ready
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "member-operator", 1, retryInterval, timeout)
+	require.NoError(t, err, "failed while waiting for operator deployment")
+
+	t.Log("member-operator is ready and running state")
+
+	extraUserAcc := createUserAccount(t, f, ctx, "amar")
+	t.Log("extra useraccount created at start")
+
+	// create useraccount
+	userAcc := newUserAcc(t, f, ctx, "johnsmith")
+	err = f.Client.Create(context.TODO(), userAcc, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(t, err)
+	t.Logf("user account '%s' created", userAcc.Name)
+
+	t.Run("verify_useraccount_ok", func(t *testing.T) {
+		err := verifyResources(t, f, userAcc)
+		assert.NoError(t, err)
+	})
+
+	t.Run("delete_user_ok", func(t *testing.T) {
+		user := &userv1.User{}
+		err := client.Get(context.TODO(), types.NamespacedName{Name: userAcc.Name}, user)
+		require.NoError(t, err)
+
+		err = client.Delete(context.TODO(), user)
+		require.NoError(t, err)
+
+		err = verifyResources(t, f, userAcc)
+		assert.NoError(t, err)
+	})
+
+	t.Run("delete_identity_ok", func(t *testing.T) {
+		identity := &userv1.Identity{}
+		err := client.Get(context.TODO(), types.NamespacedName{Name: useraccount.ToIdentityName(userAcc.Spec.UserID)}, identity)
+		require.NoError(t, err)
+
+		err = client.Delete(context.TODO(), identity)
+		require.NoError(t, err)
+
+		err = verifyResources(t, f, userAcc)
+		assert.NoError(t, err)
+	})
+
+	t.Run("delete_user_mapping_ok", func(t *testing.T) {
+		user := &userv1.User{}
+		err := client.Get(context.TODO(), types.NamespacedName{Name: userAcc.Name}, user)
+		require.NoError(t, err)
+
+		user.Identities = []string{}
+		err = client.Update(context.TODO(), user)
+		require.NoError(t, err)
+
+		err = verifyResources(t, f, userAcc)
+		assert.NoError(t, err)
+	})
+
+	t.Run("delete_identity_mapping_ok", func(t *testing.T) {
+		identity := &userv1.Identity{}
+		err := client.Get(context.TODO(), types.NamespacedName{Name: useraccount.ToIdentityName(userAcc.Spec.UserID)}, identity)
+		require.NoError(t, err)
+
+		identity.User = corev1.ObjectReference{Name: "", UID: ""}
+		err = client.Update(context.TODO(), identity)
+		require.NoError(t, err)
+
+		err = verifyResources(t, f, userAcc)
+		assert.NoError(t, err)
+	})
+
+	err = verifyResources(t, f, extraUserAcc)
+	assert.NoError(t, err)
+	t.Log("extra useraccount verified at end")
```

Comment:
I guess that this should be verified for each subtest separately. So extract it to separated function and call it at the end of each subtest

Reply:
the test_userAcc is verified after each test.  this is extra_userAcc for which we don't expect any change during sub-test execution.  so verifying at the end should be enough from my POV.